c10/core/TensorImpl.h

```diff
@@ -1496,10 +1496,13 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
    * Compute the number of elements based on the sizes of a tensor.
    */
   int64_t compute_numel() const {
-    int64_t n = 1;
-    for (auto s : sizes()) {
-      n *= s;
+    uint64_t n = 1;
+    for (uint64_t s : sizes()) {
+      uint64_t n_next = n * s;
+      TORCH_CHECK((n_next >= n) || (n_next >= s), "int64 overflow while computing numel for sizes ", sizes());
```

Comment:
&&?

Reply:
You're right. I was trying to also handle the case where either `n` or `s` is 0, but this won't do that correctly.